Pull channel and bot name from the environment

r4v5 opened the pull request:
Also, document-ish what all these environment variables actually do. I may have misinterpreted one or two of them.

r4v5 committed:
Pull channel and bot name from the environment

Also, document-ish what these environment variables actually do.

r4v5 commented:
(note that you'll probably want to set $CHANNEL and $USER in run.sh before hitting merge on this, or kaet will go down.)

rakslice commented:
I went with BOT_USER so as not to conflict with the USER variable from the OS :)

r4v5 commented:
Oh, that's a good point and we'd probably wanna prepend BOT_ on the others too.

r4v5 committed:
Preface env vars with BOT_ for cohesion

Since USER is a p. commonly-set environment variable. Update docs.
Doing them all for consistency's sake, because namespaces are cool.
